From e537726b2de79d6397b19cea7209e6df6aa890ce Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 13 Jul 2022 10:08:18 +0300
Subject: [PATCH] [SW-95553] rom: update pcie rr config

Now in rtl pif iso is implemented . According to this, by default
slave's pif is isolated . During this phase no pcie regs should be
accessed.

Hence the pcie rr configuration is separated from generic rr config
and tied along with pcie config of the respective die.

Change-Id: Iff5ca6f7e5b830d84694afc9dddaee3c38326961
---

diff --git a/arch/gaudi3/arc/dev_security.h b/arch/gaudi3/arc/dev_security.h
index 04fa2ef..1ec99ae 100644
--- a/arch/gaudi3/arc/dev_security.h
+++ b/arch/gaudi3/arc/dev_security.h
@@ -88,5 +88,6 @@
 
 int secureboot_verify_image_hash384(uint8_t *exp_hash, unsigned char *image, uint32_t image_len);
 void security_rr_recovery_config(void);
+void security_rr_recovery_pcie_config(void);
 
 #endif /* _GAUDI3_DEV_SECURITY_H_ */
diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 83656ff..91c7863 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -128,6 +128,7 @@
 	if ((arcboot_rom_global.die_id && pcie_ctrl_val) ||
 		(!arcboot_rom_global.die_id && !pcie_ctrl_val)) {
 		disable_pif_isolation();
+		security_rr_recovery_pcie_config();
 		disable_pcie_error_resp();
 		pcie_config();
 	}
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index e42944a..309f982 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -96,6 +96,7 @@
 
 static struct rom_rr_ctx rr_ctx_lbw[RR_MAX_CTX_TYPES] = {
 	[RR_OPEN_ALL] = {
+		.rr_idx = HL_SEC_LBW_RR_SHORT_13,
 		.data_min = RR_DEFAULT_VAL,
 		.data_max = RR_DEFAULT_VAL,
 		.en_val = RR_PRIV_RANGE_DIS,
@@ -109,6 +110,7 @@
 
 static struct rom_rr_ctx rr_ctx_hbw[RR_MAX_CTX_TYPES] = {
 	[RR_OPEN_ALL] = {
+		.rr_idx = HL_SEC_HBW_RR_SHORT_7,
 		.data_min = RR_DEFAULT_VAL,
 		.data_min_hi = RR_DEFAULT_VAL,
 		.data_max = RR_DEFAULT_VAL,
@@ -715,10 +717,8 @@
 }
 
 #define DEBUG_IF_ENABLE_VAL		0
-static void security_rr_config_lbw_if_common(struct rom_rr_ctx *rr_ctx)
+static void security_rr_config_lbw_debug_if(struct rom_rr_ctx *rr_ctx)
 {
-	security_rr_pcie_lbw_config(rr_ctx);
-
 	if ((arcboot_rom_global.config_fuse &
 		EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK) == DEBUG_IF_ENABLE_VAL)
 		security_rr_jtag_lbw_config(rr_ctx);
@@ -728,10 +728,8 @@
 		security_rr_twi_lbw_config(rr_ctx);
 }
 
-static void security_rr_config_hbw_if_common(struct rom_rr_ctx *rr_ctx)
+static void security_rr_config_hbw_debug_if(struct rom_rr_ctx *rr_ctx)
 {
-	security_rr_pcie_hbw_config(rr_ctx);
-
 	if ((arcboot_rom_global.config_fuse &
 		EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK) == DEBUG_IF_ENABLE_VAL)
 		security_rr_jtag_hbw_config(rr_ctx);
@@ -745,11 +743,16 @@
 {
 	security_handle_twi();
 
-	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_13;
-	security_rr_config_lbw_if_common(&rr_ctx_lbw[RR_OPEN_ALL]);
+	security_rr_config_lbw_debug_if(&rr_ctx_lbw[RR_OPEN_ALL]);
 
-	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
-	security_rr_config_hbw_if_common(&rr_ctx_hbw[RR_OPEN_ALL]);
+	security_rr_config_hbw_debug_if(&rr_ctx_hbw[RR_OPEN_ALL]);
+}
+
+void security_rr_recovery_pcie_config(void)
+{
+	security_rr_pcie_lbw_config(&rr_ctx_lbw[RR_OPEN_ALL]);
+
+	security_rr_pcie_hbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
 }
 
 void check_and_enable_4b_termination(void) {}
